Feature/add users

MonikaLukowska opened the pull request:
This PR introduce users. Implementation includes: 

- adding Devise gem
- adding User model with roles
- adding bootstrap classes to devise forms
- adding sign in, sign up, sign out links
- adding seeds
![Zrzut ekranu 2022-09-27 o 22 51 08](https://user-images.githubusercontent.com/56922072/192633001-85e1be54-10ff-4e73-a74b-e8c072dccb0b.png)
![Zrzut ekranu 2022-09-27 o 22 51 17](https://user-images.githubusercontent.com/56922072/192633048-a6d793fd-0f99-4d7c-84be-3d6bf7004e0c.png)
![Zrzut ekranu 2022-09-27 o 22 52 01](https://user-images.githubusercontent.com/56922072/192633064-c8c4e933-e07a-46be-94ea-c0558802e991.png)

MonikaLukowska committed:
chore:add devise

MonikaLukowska committed:
feat:add user

MonikaLukowska committed:
feat:add views for authentication

MonikaLukowska committed:
test:add model spec for password length

MonikaLukowska committed:
feat:add user seeds

tomdonarski commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,14 @@
+  describe 'validations' do
+    context 'when passing too long password' do
+      it 'validates password length in bytes' do
+        password = "#{'p' * 72}assword"
```
Nice but a bit opaque, and what's even more important the condition in your validator states `<= 72`.
The length of a password is rarely a business rule, so here it'd be acceptable but consider this:
Requirements live in the test suite (that is if there's some business logic - e.g. an email is to be sent exactly 48h before a movie starts - it should be covered in a test).
Given that the tests should guard us against code changes that might cause stepping over the line of requirements, the code of your validator is a bit imprecise - it introduces some play, that in case of changes to the validator will go unnoticed (someone might change it to `<= 77` and it'll pass, because the test tells that your requirement is `bytesize less than 79`).

So if the requirement is `<= 72` then I'd suggest changing it to `"x" * 73`.

MonikaLukowska replied:
Thank you for your remark, I will change it as suggested:)

tomdonarski reviewed (changes requested):
Neatly written 👏 
Please see to the remark though 🙂

h4ppyr0gu3 commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,14 @@
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    context 'when passing too long password' do
+      it 'validates password length in bytes' do
```
you might want to consider using the aggregates failure flag
```suggestion
      it 'validates password length in bytes', :aggregate_failures do
```
This will return all failures in this block with reference to what went wrong otherwise it may be difficult to find the failure with mulltiple expects in a single it

MonikaLukowska replied:
thanks for the hint, I will add it:)

MonikaLukowska committed:
test:refactor according CR remarks

change tested password length to correspond with validator; add aggregate_failure flag

tomdonarski reviewed (approved):
Now it's all good 🙂
